lib/queue.js

```diff
@@ -84,14 +84,20 @@ util.inherits(KafkaProducer, Producer);
 /** @inheritdoc */
 KafkaProducer.prototype.send = function(batch) {
     var self = this;
-    // Use JSON serialization
-    batch.messages = batch.messages.map(JSON.stringify);
-    // Enable snappy compression
-    batch.attributes = 2;
-    return this.ready.then(function() {
-        self.logger.log('trace/kafka/send', batch);
-        return self.producer.sendAsync(batch);
+    if (!Array.isArray(batch)) {
```

Comment:
Should we support passing in something that is not an array? It sounds like a potential bug in the caller.